plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -636,6 +664,162 @@ public void testRegrAggregationPushdown()
         }
     }
 
+    @Test
+    public void testJoinPushdownDisabled()
+    {
+        // If join pushdown gets enabled by default, this test should use a session with join pushdown disabled
+        Session noJoinPushdown = Session.builder(getSession())
+                // Disable dynamic filtering so that expected plans in case of no pushdown remain "simple"
+                .setSystemProperty("enable_dynamic_filtering", "false")
+                // Disable optimized hash generation so that expected plans in case of no pushdown remain "simple"
+                .setSystemProperty("optimize_hash_generation", "false")
+                .build();
+
+        PlanMatchPattern partitionedJoinOverTableScans = node(JoinNode.class,
+                exchange(ExchangeNode.Scope.REMOTE, ExchangeNode.Type.REPARTITION,
+                        node(TableScanNode.class)),
+                exchange(ExchangeNode.Scope.LOCAL, ExchangeNode.Type.REPARTITION,
+                        exchange(ExchangeNode.Scope.REMOTE, ExchangeNode.Type.REPARTITION,
+                                node(TableScanNode.class))));
+
+        assertThat(query(noJoinPushdown, "SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey = r.regionkey"))
+                .isNotFullyPushedDown(partitionedJoinOverTableScans);
+    }
+
+    @Test
+    public void testJoinPushdown()
```

Comment:
It looks like a feature is written in a generic way, while testing only happens for PostgreSQL. Can we somehow disable this in other connectors until the moment test coverage is added to them?

Reply:
it's disabled by default, and i am still thinking how to apply the test to all connectors, sanely.